db/background_mgr_resync_dcp.go

```diff
@@ -242,6 +242,22 @@ func (r *ResyncManagerDCP) Run(ctx context.Context, options map[string]interface
 		base.InfofCtx(ctx, base.KeyAll, "[%s] resync was terminated. Docs changed: %d Docs Processed: %d", resyncLoggingID, r.DocsChanged.Value(), r.DocsProcessed.Value())
 	}
 
+	collectionsRequiringResync := make([]base.ScopeAndCollectionName, 0)
```

Comment:
We should only do this work if we successfully updated SyncInfo for any collections (on line 223) - otherwise can skip this (which would include any resync done with regenerate_sequences=false)

Reply:
Actually, can we just remove any collections that we successfully updated on line 223 from db.RequireResync, instead of recomputing the set?

Reply:
This is much easier to reason through, addressed.